src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -184,7 +185,7 @@ module.exports = function<T, P, I, TI, C>(
     }
   }
 
-  function unmountHostComponents(parent, current): void {
+  function unmountHostComponents(parent, current, ignoreHostNode, ignoreTop): void {
```

Comment:
I think we can find a way to not use these flags.

A normal unmount won't schedule a removeChild anywhere but the top root where it was deleted. All we have to do is special case the root and then the rest of the traversal can be exactly the same.

Similarly, ignoreTop wouldn't be needed if instead we did the root work outside this function instead of inside it. That way we don't have to branch inside the loop.